server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -1051,6 +1053,7 @@ public ClusterState execute(ClusterState currentState) {
                 boolean changed = false;
                 for (RestoreInProgress.Entry entry : currentState.custom(RestoreInProgress.TYPE, RestoreInProgress.EMPTY)) {
                     if (entry.state().completed()) {
+                        logger.log(entry.silent() ? Level.DEBUG : Level.INFO, "restore of snapshot [{}] completed", entry.snapshot());
```

Comment:
@fcofdez suggested to me this is a nice place to log the completion of a snapshot restore. This meant passing the silent property from the `RestoreSnapshotRequest` to `RestoreInProgress.Entry` so that we can read the silent parameter here.